src/views/virtualmachines/tree/utils/utils.tsx

```diff
@@ -191,25 +248,32 @@ const getVMInfoFromPathname = (pathname: string) => {
   return { currentVMTab, vmName, vmNamespace };
 };
 
-export const createTreeViewData = (
+const createClusterTreeViewData = (
   getResourceUrl: GetResourceUrl,
   projectNames: string[],
   vms: V1VirtualMachine[],
   isAdmin: boolean,
   pathname: string,
-  foldersEnabled: boolean,
+  treeViewDataMap: Record<string, TreeViewDataItem> = {},
+  clusterProjectMap: Record<string, any> = {},
+  clusterName?: string,
 ): TreeViewDataItem[] => {
-  const { currentVMTab, vmName, vmNamespace } = getVMInfoFromPathname(pathname);
-
-  const treeViewDataMap: Record<string, TreeViewDataItem> = {};
-  const projectMap = buildProjectMap(
-    getResourceUrl,
-    vms,
-    vmName,
-    currentVMTab,
-    treeViewDataMap,
-    foldersEnabled,
-  );
+  const { vmName, vmNamespace } = getVMInfoFromPathname(pathname);
+
+  const projectMap = {};
+
+  // assumes cluster name will always be unique
+  Object.keys(clusterProjectMap).forEach((key) => {
+    if (!clusterName) {
+      const mappedProjectName = key;
+      projectMap[mappedProjectName] = clusterProjectMap[key];
+    } else {
+      const [mappedClusterName, mappedProjectName] = key.split('/');
+      if (clusterName == mappedClusterName) {
+        projectMap[mappedProjectName] = clusterProjectMap[key];
+      }
+    }
+  });
```

Comment:
Let's structure this in a way that allows project map to work in both cases without having to rebuild the project map. 
This may require a two level map, first level could be clusters, second could be projects. 

Key having a slash in it could still work here. Maybe use the format in both cases and use a single-cluster key at the front of the slash key to id the single cluster case.